test/Main.hs

```diff
@@ -131,9 +131,34 @@ main = do
                 propsChunk config
                 propsChunk2 config
                 propsBig config 512 5 largeHC
+            describe "addEndMark 0 defaultConfig" $ do
+                -- 0 is not a good end mark then we don't removeUncompressed
+                let config = addEndMark (-1) defaultConfig
```

Comment:
Why even have any Endmark other than 0?
 
When we have uncompressed length in the header then we should have the compressed length first and then uncompressed length. So there won't be any confusion between endmark and block header.